src/parsers.rs

```diff
@@ -164,6 +164,20 @@ impl<'a> Parser<'a> {
             }
         };
 
+        // Check for an IPv4 mapped address
+        if *head_idx == 6 || use_tail && *tail_idx == 1 {
```

Comment:
Shouldn't this also check that the current `u16` is `0xffff`?

Reply:
yes, fixed.